### PR TITLE
Warn about missing phone number

### PR DESCRIPTION
**Pull request by reynders**
#### Summary
- Warn about missing phone number
- Alert supports one or two alerts
- E2E tests

Screenshot 2022-01-17 at 11 33 46

### CODE REVIEW

File: frontend/src/citizen-frontend/personal-details/PersonalDetails.tsx [hunk 1]
```diff
@@ -123,6 +123,12 @@ export default React.memo(function PersonalDetails() {
                     data-qa="missing-email-box"
                   />
                 )}
+                {!phone && (
+                  <AlertBox
+                    message={t.personalDetails.noPhoneAlert}
+                    data-qa="missing-phone-box"
+                  />
+                )}
```
[the remark this answers is not in the document]
Comment (reynders) [reply frontend/src/citizen-frontend/personal-details/PersonalDetails.tsx]: Done

[the remark this answers is not in the document]
Comment (reynders) [reply frontend/src/citizen-frontend/personal-details/PersonalDetails.tsx]: Screenshot above

File: frontend/src/citizen-frontend/personal-details/PersonalDetails.tsx [hunk 2]
```diff
@@ -117,10 +117,13 @@ export default React.memo(function PersonalDetails() {
 
             return (
               <>
-                {email === null && (
+                {(!email || !phone) && (
```
Comment (jaahon) [inline frontend/src/citizen-frontend/personal-details/PersonalDetails.tsx:120]: Now both of the warnings are shown even if one of them is already filled.

File: frontend/src/citizen-frontend/personal-details/PersonalDetails.tsx [hunk 3]
```diff
@@ -117,10 +117,13 @@ export default React.memo(function PersonalDetails() {
...
                   <AlertBox
-                    message={t.personalDetails.noEmailAlert}
-                    data-qa="missing-email-box"
+                    message={[
+                      t.personalDetails.noEmailAlert,
+                      t.personalDetails.noPhoneAlert
+                    ].join('\n')}
```
Comment (jaahon) [inline frontend/src/citizen-frontend/personal-details/PersonalDetails.tsx:125]: This does not actually add a line break in the text. You need a `<br>` element or wrap the sentences in their own `<p>` elements.

File: frontend/src/citizen-frontend/personal-details/PersonalDetails.tsx (hunk 2 above)
Comment (reynders) [reply frontend/src/citizen-frontend/personal-details/PersonalDetails.tsx:120]: The alert is shown if either email or phone is missing, and the alert text reflects dynamically what is missing?

Comment (reynders) [reply frontend/src/citizen-frontend/personal-details/PersonalDetails.tsx:120]: 🤦 got it, did two refactoring that cancelled each other out

File: frontend/src/citizen-frontend/personal-details/PersonalDetails.tsx (hunk 3 above)
Comment (reynders) [reply frontend/src/citizen-frontend/personal-details/PersonalDetails.tsx]: The alert element (and the stack of component it uses) does not support this, so text formatting is used instead

**Review by jaahon — APPROVED** [review]

**Pull request merged**